### ISSUE TITLE
Trying to get Teensy3 (with Teensyduino) working with recent Arduino releases

### ISSUE THREAD

**Issue by blackketter** [xxxajk/Arduino_Makefile_master#14]
This project looks awesome, but I'm stuck.

Using Arduino 1.6.5 or 1.6.6 is not working for me, I have to define many variables in my makefile and make some minor changes to the source.  What's the latest supported version?  Am I missing something basic here?

Thanks in advance.

My Makefile so far is, which seems unnecessary complex:

BOARD_TAG    = teensy31
ARDUINO_PLATFORM_LIB_PATH = /Applications/Arduino.app/Contents/Java/hardware/teensy/avr/libraries
ARCHITECTURE = arm
ARDUINO_ARCH_FLAG = -DARDUINO_ARCH_AVR

MCU = cortex-m4
F_CPU = 72000000
USB_TYPE = USB_SERIAL_HID

LINKSCRIPT = mk20dx256.ld

CPPFLAGS = -mthumb -D__MK20DX256__ -DTEENSYDUINO=126

CXXFLAGS_STD += -std=gnu++0x
CXXFLAGS += -felide-constructors 
CXXFLAGS += -fno-rtti
CXXFLAGS += -fsingle-precision-constant
CXXFLAGS += -M -MG -MP -w -x c++

LDFLAGS = -mthumb
LDFLAGS += -fsingle-precision-constant
LDFLAGS += -larm_cortexM4l_math

include ~/Documents/Arduino/Arduino-Makefile/Teensy.mk

**Comment by xxxajk** [comment]
Correct. It does not do the new schema correctly yet, but it will soon.

**Comment by blackketter** [comment]
Ah, thanks.  Please let me know if there's anything I can do to help.

**Comment by xxxajk** [comment]
I'm considering to scrape the data for 1.6.6+ using the arduino CLI build tool. The CLI tool is quite flexible.
This gives two options:
1: use make for everything still, but use arduino builder to gather information only
2: use arduino builder for the actual build.

arduino builder is actually pretty quick.

**Comment by blackketter** [comment]
I'm finding that the builds are slower on 1.6.6 than 1.6.5 and that there's a significant amount of compiling of files that haven't changed.  Not sure where the root of this problem is, but if make can be wiser about unnecessary rebuilds I'd vote for that.

**Comment by xxxajk** [comment]
Eventually this will happen with a cache using the makefile system, and keyed to a hash of the command line passed to the OS. That way things like core components don't get rebuilt, and instead get copied or symlinked.
